dora-metrics/pelorus
# Fix: Remove symbolic link from charts folder

## Linked Issues

No.

## Description

Remove unnecessary symbolic link from charts folder.

To help me in the changes, I have run
```sh
grep -Iinr 'subcharts' . --exclude-dir=.venv
```

## Testing Instructions

### normal helm installation

To test normal helm installation, run
```sh
oc create namespace test-pelorus-helm
helm install operators charts/operators --namespace test-pelorus-helm
helm install pelorus charts/pelorus --namespace test-pelorus-helm
```
and check the 3 exporters `/metrics` endpoint and Grafana.

To clean it up, run
```sh
helm uninstall pelorus --namespace test-pelorus-helm
helm uninstall operators --namespace test-pelorus-helm
```

### separate helm installation

To test separate helm installation, run
```sh
oc create namespace test-pelorus-helm-separate
cat > pelorus-sample-separate-exporters.yaml <<EOF
instances:
  - app_name: deploytime-exporter
    exporter_type: deploytime
  - app_name: failuretime-exporter
    enabled: false
    exporter_type: failure
  - app_name: committime-exporter
    exporter_type: committime
  - app_name: webhook-exporter
    exporter_type: webhook
EOF
helm install custom-exporters-deployment charts/pelorus/charts/exporters --namespace test-pelorus-helm-separate --values pelorus-sample-separate-exporters.yaml
```
and check the 3 exporters `/metrics` endpoint.

To clean it up, run
```sh
helm uninstall custom-exporters-deployment --namespace test-pelorus-helm-separate
```

### operator

To test operator, run
```sh
operator-sdk run bundle \
quay.io/pelorus/rc-pelorus-operator-bundle:vpr969-abc123 \
--namespace test-pelorus-operator
```
and check the 3 exporters `/metrics` endpoint and Grafana.

To clean it up, run
```sh
operator-sdk cleanup pelorus-operator --namespace test-pelorus-operator
```

It was generated running
```sh
./scripts/create_pelorus_operator.sh -f -v 0.0.7-rc.1

export REPOSITORY=quay.io/pelorus
export pr_type=opened
export pr_number=969
export commit_hash=abc123

cd pelorus-operator
export TEST_VERSION=pr$pr_number-$commit_hash
export CURRENT_OPERATOR_VERSION=$(grep "VERSION ?= " Makefile  | cut -c 12-)
export CURRENT_CHART_VERSION="$(grep '^version: ' helm-charts/pelorus/Chart.yaml  | cut -c 10-)"
export DEFAULT_OPERATOR_IMAGE=$REPOSITORY/pelorus-operator:$CURRENT_OPERATOR_VERSION
export TEST_OPERATOR_IMAGE=$REPOSITORY/rc-pelorus-operator:$TEST_VERSION

sed -i "s,$DEFAULT_OPERATOR_IMAGE,$TEST_OPERATOR_IMAGE,g" bundle/manifests/pelorus-operator.clusterserviceversion.yaml
sed -i "s,$CURRENT_CHART_VERSION,$TEST_VERSION,g" helm-charts/pelorus/charts/exporters/templates/_deploymentconfig.yaml
sed -i "s,$CURRENT_OPERATOR_VERSION,$TEST_VERSION,g" Makefile
find . -type f | xargs sed -i "s,$CURRENT_OPERATOR_VERSION,$CURRENT_OPERATOR_VERSION-$TEST_VERSION,g"
find . -type f | xargs sed -i "s,$CURRENT_CHART_VERSION,$CURRENT_CHART_VERSION-$TEST_VERSION,g"
sed -i "s,pelorus-operator,rc-pelorus-operator,g" Makefile

helm dep update helm-charts/pelorus
rm -r helm-charts/pelorus/charts/*.tgz

make podman-build
make bundle-build
make podman-push
make bundle-push
```
in this branch.
